### PR TITLE
[UNR-5559] Fix: create snapshot folder at start of deployment, delete if empty for interop with squid snapshot endpoint

### PR DESCRIPTION
**Pull request by simonsarginson**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed an issue that caused `UnrealGDK/Setup.sh` to report `sed: can't read : No such file or directory` when run on macOS.
 - Static subobjects on bNetLoadOnClient actors are now removed on clients in a manner matching native unreal's behavior. This change affects subobjects removed by the server while the actor is not in the client's interest.
 - Fixed an issue where multicast rpcs could be overwritten and then dropped on authority flicker.
+- Fixed issue using squid snapshot end point with a local deployment, using `localhost:5006/snapshot` works again and creates a snapshot.
```
Comment (joshuahuburn) [inline CHANGELOG.md:45]: Squid is not publicly known. Please just refer to as runtime.

Comment (simonsarginson) [reply CHANGELOG.md:45]: ok didn't know that.

Comment (simonsarginson) [inline CHANGELOG.md:45]:
```suggestion
- Fixed issue using the runtime snapshot endpoint with a local deployment, using `localhost:5006/snapshot` works again and creates a snapshot.
```

**Review by joshuahuburn — APPROVED** [review]
Looks good

File: SpatialGDK/Source/SpatialGDKServices/Public/LocalDeploymentManager.h [hunk 2]
```diff
@@ -111,6 +111,8 @@ class FLocalDeploymentManager final
 
 	FString ExposedRuntimeIP;
 
+	FString LastSnapshotPath;
```
Comment (KeyMaster-) [inline SpatialGDK/Source/SpatialGDKServices/Public/LocalDeploymentManager.h:114]:
```suggestion
	FString CurrentSnapshotDirectory;
```
`LastSnapshotPath` reads to me like the path to the latest snapshot that was taken.

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 3]
```diff
@@ -495,6 +498,21 @@ void FLocalDeploymentManager::FinishLocalDeploymentShutDown()
 	// Kill the log file handle.
 	RuntimeLogFileHandle.Reset();
 
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	const auto IsDirectoryEmpty = [&PlatformFile](const TCHAR* Directory) -> bool {
+		bool directoryIsEmpty = true;
```
Comment (KeyMaster-) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp:503]:
```suggestion
		bool bIsDirectoryEmpty = true;
```

**Review by KeyMaster- — APPROVED** [review]
Some naming suggestions, looks good from a functionality side 👍

File: SpatialGDK/Source/SpatialGDKServices/Public/LocalDeploymentManager.h (hunk 2 above)
Comment (simonsarginson) [reply SpatialGDK/Source/SpatialGDKServices/Public/LocalDeploymentManager.h:114]: Sure I'm fine with that.

**Pull request merged**